### PR TITLE
[Broker] modify permission in metadata RocksdbMetadataStore

### PR DESCRIPTION
**Pull request by skysiders**
PULSAR-18 modify permission in RocksdbMetadataStore

### Motivation

While RocksdbMetadataStore init , it will create directory from dataPath use Files.createDirectories(dataPath); this path create with too loose that any one can read file in this directory.This may cause everyone can read logs, perhaps strict permission is more safe.

### Modifications

set permission for dataPath.

- [x] `no-need-doc` 
this is just a permission update

### CODE REVIEW

**Comment by skysiders** [comment]
In this patch, I set 700 permission for data path and log path.

File: pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/RocksdbMetadataStore.java [hunk 1]
```diff
@@ -220,6 +223,8 @@ private RocksdbMetadataStore(String metadataURL, MetadataStoreConfig metadataSto
         Path dataPath = FileSystems.getDefault().getPath(dataDir);
         try {
             Files.createDirectories(dataPath);
+            Set<PosixFilePermission> perms = PosixFilePermissions.fromString("rwx------");
+            Files.setPosixFilePermissions(dataPath, perms);
         } catch (IOException e) {
             e.printStackTrace();
```
Comment (Technoboy-) [inline pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/RocksdbMetadataStore.java:229]: Could you help to change `log.error` ?

File: pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/RocksdbMetadataStore.java [hunk 2]
```diff
@@ -220,6 +223,8 @@ private RocksdbMetadataStore(String metadataURL, MetadataStoreConfig metadataSto
         Path dataPath = FileSystems.getDefault().getPath(dataDir);
         try {
             Files.createDirectories(dataPath);
+            Set<PosixFilePermission> perms = PosixFilePermissions.fromString("rwx------");
```
Comment (Jason918) [inline pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/RocksdbMetadataStore.java:226]: "rw-r-----" is better?

File: pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/RocksdbMetadataStore.java [hunk 3]
```diff
@@ -313,6 +318,8 @@ private void configLog(Options options) throws IOException {
             logPathSetting = FileSystems.getDefault().getPath(logPath + "/rocksdb-log");
             Files.createDirectories(logPathSetting);
             options.setDbLogDir(logPathSetting.toString());
+            Set<PosixFilePermission> perms = PosixFilePermissions.fromString("rwx------");
```
Comment (Jason918) [inline pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/RocksdbMetadataStore.java:321]: Same here.

File: pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/RocksdbMetadataStore.java (hunk 1 above)
Comment (Jason918) [reply pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/RocksdbMetadataStore.java:229]: > Could you help to change to `log.error` ?

We should throw exception here. I will fix this.

File: pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/RocksdbMetadataStore.java (hunk 2 above)
Comment (skysiders) [reply pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/RocksdbMetadataStore.java:226]: I think directory should have x permission here

File: pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/RocksdbMetadataStore.java (hunk 3 above)
Comment (skysiders) [reply pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/RocksdbMetadataStore.java:321]: Hi @Jason918 , this fix and line 226 are both set permission for directory, perhaps "rwxr-x---" is better? And for file "rw-r-----" is better.

Comment (Jason918) [reply pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/RocksdbMetadataStore.java:321]: "rwxr-x---" +1

Comment (skysiders) [reply pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/RocksdbMetadataStore.java:321]: Thanks for review, and I found some permission problem for PULSAR-19 and PR for #15411 ,can you please take a look?

**Review by Jason918 — APPROVED** [review]
LGTM

**Review by mattisonchao — APPROVED** [review]

**Comment by mattisonchao** [comment]
@skysiders  We should fix the checkstyle issue.

```java
Error:  src/main/java/org/apache/pulsar/metadata/impl/RocksdbMetadataStore.java:[70,1] (imports) ImportOrder: Extra separation in import group before 'java.nio.file.attribute.PosixFilePermission'
Error:  src/main/java/org/apache/pulsar/metadata/impl/RocksdbMetadataStore.java:[70,1] (imports) ImportOrder: Import java.nio.file.attribute.PosixFilePermission appears after other imports that it should precede
```

**Comment by skysiders** [comment]
Hi @mattisonchao , thanks for your review, I use check style to fix style issue, but perhaps some problem in my environment, I get `[WARNING] Unable to locate Source XRef to link to - DISABLED` after run `mvn checkstyle:checkstyle` so I check style by myself until get no error after run `mvn checkstyle:checkstyle`.

**Comment by skysiders** [comment]
> Hi @mattisonchao , thanks for your review, I use check style to fix style issue, but perhaps some problem in my environment, I get `[WARNING] Unable to locate Source XRef to link to - DISABLED` after run `mvn checkstyle:checkstyle` so I check style by myself until get no error after run `mvn checkstyle:checkstyle`.

I use `mvn checkstyle:check` solve it.

**Review by HQebupt — APPROVED** [review]

**Comment by skysiders** [comment]
Hi @Technoboy- ,could you please have a look at this ?

**Review by Technoboy- — APPROVED** [review]

**Pull request merged**